compiler/haskell-ide-core/src/Development/IDE/State/Shake.hs

```diff
@@ -119,7 +117,7 @@ getIdeGlobalState = getIdeGlobalExtras . shakeExtras
 type Values =
     Map.HashMap FilePath
```

Comment:
Given we are now just storing values wouldn't Map (FilePath, Key) be better? The nested structure was only to get diagnostics.

Reply:
Yeah, that makes sense, we key stuff by file for garbage collection, but that shouldn't be too much trouble